caches/path-mapped/src/main/java/org/commonjava/maven/galley/cache/pathmapped/PathMappedCacheProvider.java

```diff
@@ -229,47 +209,24 @@ public String getStoragePath( final ConcreteResource resource )
     public Transfer getTransfer( final ConcreteResource resource )
     {
         Transfer txfr = new Transfer( resource, this, fileEventManager, transferDecorator );
-
-        final int timeoutSeconds = resource.getLocation()
-                                           .getAttribute( Location.CACHE_TIMEOUT_SECONDS, Integer.class,
-                                                          config.getDefaultTimeoutSeconds() );
-
+        final int timeoutSeconds = getResourceTimeoutSeconds( resource );
         if ( !resource.isRoot() && config.isTimeoutProcessingEnabled() && timeoutSeconds > 0 )
         {
-            if ( isFileTimeout( txfr, timeoutSeconds ) )
-            {
-                toDelete.add( txfr );
-            }
+            deleteExecutor.submit( () -> {
```

Comment:
I'm a little curious here. Seems this may cause a not existed transfer returning if it is timeout. Am I right?

Reply:
The isFileTimeout has a DB access so I thread it off. If the file not exist, the method will return -1.